ctru-rs/src/services/ps.rs

```diff
@@ -83,3 +83,59 @@ impl Drop for Ps {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use super::*;
+
+    #[test]
+    fn construct_hash_map() {
+        let _ps = Ps::init().unwrap();
+
+        let mut input = vec![
+            (1_i32, String::from("123")),
+            (2, String::from("2")),
+            (6, String::from("six")),
+        ];
+
+        let map: HashMap<i32, String> = HashMap::from_iter(input.clone());
+
+        let mut actual: Vec<_> = map.into_iter().collect();
+        input.sort();
+        actual.sort();
+
+        assert_eq!(input, actual);
+    }
+
+    #[test]
+    fn construct_hash_map_no_rand() {
+        // Without initializing PS, we can't use `libc::getrandom` and constructing
+        // a HashMap panics at runtime.
+        //
+        // If any test case successfully creates a HashMap before this test,
+        // the thread-local RandomState in std will be initialized. We spawn
+        // a new thread to actually create the hash map, since even in multi-threaded
+        // test environment there's a chance this test wouldn't panic because
+        // some other test case ran before it.
+        //
+        // One downside of this approach is that the panic handler for the panicking
+        // thread prints to the console, which is not captured by the default test
+        // harness and prints even when the test passes.
+        crate::thread::Builder::new()
+            .stack_size(0x20_0000)
+            .spawn(|| {
+                let map: HashMap<i32, String> = HashMap::from_iter([
+                    (1_i32, String::from("123")),
+                    (2, String::from("2")),
+                    (6, String::from("six")),
+                ]);
+
+                dbg!(map);
+            })
+            .unwrap()
+            .join()
+            .expect_err("should have panicked");
```

Comment:
This is a bit weird, hence the large comment, and relies on some implementation details of `std`, which I don't love. Here is the output I was referencing:


![tests_Pass](https://user-images.githubusercontent.com/11131775/152889780-682df70d-dc09-4af8-8456-c2f7bdbf826c.png)

Perhaps it makes more sense to remove this test case entirely, since we don't necessarily care if `HashMap::new` panics, but we do want the positive test case of being able to construct one when we have initialized PS. Let me know what you think or if you have a cleaner suggestion.

Reply:
Looks like this will be fixed once we move to std threads: https://github.com/rust-lang/rust/pull/78227

I think we should keep this test regardless.